docs: Add note about IAM roles for google.batch.serviceAccountEmail (#7291)
nextflow-io/nextflow: docs/reference/config.mdx

##### `google.batch.serviceAccountEmail`

The Google service account email to use for the pipeline execution. If not specified, the default Compute Engine service account for the project will be used.

> The service account must have the IAM roles listed in the [Google Cloud Batch documentation for Nextflow](https://cloud.google.com/batch/docs/nextflow), otherwise Cloud Batch jobs can remain stuck in the `SCHEDULED` state indefinitely.

Reviewer: ```suggestion
The service account must have the IAM roles listed in the [Google Cloud Batch documentation for Nextflow](https://cloud.google.com/batch/docs/nextflow). Without these roles, Cloud Batch jobs stay in the `SCHEDULED` state and never run.
```
